src/components/TagManagementDialog/AddTagForm.tsx

```diff
@@ -0,0 +1,214 @@
+		const failedRule = newTagValidationRules<T>(t, key, existingTags).find(
+			(rule) => rule.test(),
+		);
+
+		const hasErrors = !!failedRule;
+
+		setTagKeyIsInvalid(hasErrors);
+		setError(failedRule);
+		setCanSubmit(!hasErrors);
+		return hasErrors;
+	};
+
+	const checkTagOverwrites = async () => {
+		const overridableTag = find(overridableTags, {
+			tag_key: tagKey,
+		});
+
+		if (!overridableTag) {
+			return true;
+		}
+		const count = overridableTag.items.length;
+		const result = await new Promise<boolean>((resolve) => {
+			const confirmationOptions = {
+				title: t('warnings.this_would_overwrite_tags', {
+					itemType,
+					count,
+				}),
+				children: (
+					<Typography>
+						{t(`warnings.tag_name_group_exists_and_will_be_overwritten`, {
+							itemType,
+							count,
+						})}
+						<br />
+						{t('actions_confirmations.confirm_to_proceed')}
+					</Typography>
+				),
+				action: t('actions.continue'),
+				onClose: resolve,
+			} as SimpleConfirmationDialogProps;
+
+			setConfirmationDialogOptions(confirmationOptions);
+		});
+		setConfirmationDialogOptions(undefined);
+		return result;
+	};
+
+	const internalAddTag = withPreventDefault(() => {
+		if (checkNewTagValidity(tagKey)) {
+			return;
+		}
+
+		return checkTagOverwrites().then((confirmed) => {
+			if (!confirmed) {
+				return;
+			}
+
+			addTag({
+				tag_key: tagKey,
+				value,
+			} as ResourceTagInfo<T>);
+
+			setTagKey('');
+			setValue('');
+			setTagKeyIsInvalid(false);
+			setError(undefined);
+			setCanSubmit(false);
+
+			if (tagKeyInput?.current) {
+				tagKeyInput.current.blur();
+			}
+			if (valueInput?.current) {
+				valueInput.current.blur();
+			}
+		});
+	});
+
+	return (
+		<>
+			<Stack
+				gap={2}
+				mb={2}
+				alignItems="center"
+				direction="row"
+				onKeyDown={(e) => {
+					stopKeyDownEvent(e, 13, internalAddTag);
+				}}
+			>
+				<TextField
+					inputProps={{
+						form: formUuid,
+					}}
+					fullWidth
+					ref={tagKeyInput}
+					onChange={(e) => {
+						setTagKey(e.target.value);
+						checkNewTagValidity(e.target.value);
+					}}
+					value={tagKey}
+					error={tagKeyIsInvalid}
+					placeholder={t('labels.tag_name')}
+				/>
+
+				<TextField
+					inputProps={{
+						form: formUuid,
+					}}
+					fullWidth
+					ref={valueInput}
+					onChange={(e) => {
+						setValue(e.target.value);
+					}}
+					value={value}
+					placeholder={t('labels.value')}
+				/>
+
+				<form id={formUuid} onSubmit={internalAddTag}>
+					<Button
+						sx={{
+							width: 120,
+							// TODO: check what tertiary replacement is
+							color: designTokens.color.palette.neutral[100],
+						}}
+						onClick={internalAddTag}
+						disabled={!canSubmit}
+					>
+						{t('actions.add_tag')}
+					</Button>
+				</form>
+				{confirmationDialogOptions && (
+					<SimpleConfirmationDialog {...confirmationDialogOptions} />
+				)}
+			</Stack>
+			<Box>{error && <Callout severity="danger">{error.message}</Callout>}</Box>
```

Comment:
What is the Box there for? Can we remove it so it's not in the DOM when there is no error?